src/invocation/ClientConnection.ts

```diff
@@ -42,14 +48,23 @@ class ClientConnection {
     connect(): Promise<ClientConnection> {
         var ready = Promise.defer<ClientConnection>();
 
-        this.socket = net.connect(this.address.port, this.address.host, () => {
-
+        let conCallback = () => {
             // Send the protocol version
             var buffer = new Buffer(3);
             buffer.write('CB2');
             this.socket.write(buffer);
             ready.resolve(this);
-        });
+        };
+
+        if (this.clientNetworkConfig.sslOptions) {
+            let sslSocket = tls.connect(this.address.port, this.address.host, this.clientNetworkConfig.sslOptions, conCallback);
```

Comment:
There is a catch about `let` keyword. Node.js 4.x does not support `let` so our Typescript setup is configured to translate `let` keywords to `var` anyway. Since `let` is block-scoped and `var` is not, use of `let` may fool developers. Also this may cause hard to spot anomalies in behavior of the client if we move to newer versions of Node.js in future. So I prefer not to use `let` at all.